src/core/qgsrendercontext.cpp

```diff
@@ -359,3 +384,34 @@ double QgsRenderContext::convertFromMapUnits( double sizeInMapUnits, QgsUnitType
   }
   return 0.0;
 }
+
+double QgsRenderContext::convertMetersToMapUnits( double meters ) const
+{
+  switch ( mDistanceArea.sourceCrs().mapUnits() )
+  {
+    case QgsUnitTypes::DistanceMeters:
+      return meters;
+    case QgsUnitTypes::DistanceDegrees:
+    {
+      QgsPointXY pointCenter = mExtent.center();
+      // The Extent is in the sourceCrs(), when different from destinationCrs()
+      // - the point must be transformed, since DistanceArea uses the destinationCrs()
+      // Note: the default QgsCoordinateTransform() : authid() will return an expty String
+      if ( mCoordTransform.sourceCrs().authid() != mCoordTransform.destinationCrs().authid() )
```

Comment:
Potential speed bump - you could use

    if ( !mCoordTransform.isShortCircuited() )

here instead. It's effectively the same check (source != dest), but calculated in advance so it avoids the need for a string comparison with every call.
Alternatively QgsCoordinateTransform::transform already has this check built in, so you could just avoid the check and always call `pointCenter = mCoordTransform.transform(...)`. The transform is automatically skipped if not required.